extensions/data-workspace/src/dialogs/newProjectDialog.ts

```diff
@@ -0,0 +1,151 @@
+import { IWorkspaceService } from '../common/interfaces';
+import * as constants from '../common/constants';
+import { IProjectType } from 'dataworkspace';
+import { directoryExist } from '../common/utils';
+
+class NewProjectDialogModel {
+	projectTypeId: string = '';
+	projectFileExtension: string = '';
+	name: string = '';
+	location: string = '';
+}
+export class NewProjectDialog extends DialogBase {
+	private model: NewProjectDialogModel = new NewProjectDialogModel();
+	constructor(private workspaceService: IWorkspaceService) {
+		super(constants.NewProjectDialogTitle, 'NewProject');
+	}
+
+	async validate(): Promise<boolean> {
+		try {
+			// the selected location should be an existing directory
+			const parentDirectoryExists = await directoryExist(this.model.location);
+			if (!parentDirectoryExists) {
+				this.showErrorMessage(constants.ProjectParentDirectoryNotExistError(this.model.location));
+				return false;
+			}
+
+			// there shouldn't be an existing sub directory with the same name as the project in the selected location
+			const projectDirectoryExists = await directoryExist(path.join(this.model.location, this.model.name));
+			if (projectDirectoryExists) {
+				this.showErrorMessage(constants.ProjectDirectoryAlreadyExistError(this.model.name, this.model.location));
+				return false;
+			}
+
+			return true;
+		}
+		catch (err) {
+			this.showErrorMessage(err?.message ? err.message : err);
+			return false;
+		}
+	}
+
+	async onComplete(): Promise<void> {
+		try {
+			await this.workspaceService.createProject(this.model.name, vscode.Uri.file(this.model.location), this.model.projectTypeId);
+		}
+		catch (err) {
+			vscode.window.showErrorMessage(err?.message ? err.message : err);
+		}
+	}
+
+	protected async initialize(view: azdata.ModelView): Promise<void> {
+		const allProjectTypes = await this.workspaceService.getAllProjectTypes();
+		const projectTypeRadioCardGroup = view.modelBuilder.radioCardGroup().withProperties<azdata.RadioCardGroupComponentProperties>({
+			cards: allProjectTypes.map((projectType: IProjectType) => {
+				return <azdata.RadioCard>{
+					id: projectType.id,
+					label: projectType.displayName,
+					icon: projectType.icon,
+					descriptions: [
+						{
+							textValue: projectType.displayName,
+							textStyles: {
+								'font-size': '13px',
+								'font-weight': 'bold'
+							}
+						}, {
+							textValue: projectType.description
+						}
+					]
+				};
+			}),
+			iconHeight: '25px',
+			iconWidth: '25px',
+			cardWidth: '250px',
+			cardHeight: '130px',
+			ariaLabel: constants.ProjectTypeSelectorTitle,
+			width: '500px',
+			iconPosition: 'left',
+			selectedCardId: allProjectTypes.length > 0 ? allProjectTypes[0].id : undefined
+		}).component();
+
+		this.register(projectTypeRadioCardGroup.onSelectionChanged((e) => {
+			this.model.projectTypeId = e.cardId;
+		}));
+
+		const projectNameTextBox = view.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
```

Comment:
should there be an info icon that says that a new folder will get created in the selected directory with the project name?

Reply:
this is the same experience as the current sql-db-proj extension.